packages/ra-ui-materialui/src/input/DateInput.js

```diff
@@ -41,59 +42,63 @@ const sanitizeValue = value => {
     return dateFormatter(new Date(value));
 };
 
-export class DateInput extends Component {
-    onChange = event => {
-        this.props.input.onChange(event.target.value);
+export const DateInput = ({
+    className,
+    meta,
+    input,
+    isRequired,
+    label,
+    options,
+    source,
+    resource,
+    helperText,
+    ...rest
+}) => {
+    const handleChange = event => {
```

Comment:
please use useCallback to memoize the function and avoid rerenderings of TextField